misc/determine-version.py

```diff
@@ -191,17 +211,23 @@ def version_cmp(a, b):
 all_tags = sorted(all_tags, cmp=version_cmp, reverse=True)
 verbose_print("all_tags   =", all_tags)
 
-if len(all_tags) > 1:
-    # This is a new minor version
-    final_print("%s.%d.0a.%s" % (all_tags[0][0], int(all_tags[0][1]) + 1, abbrev_rev))
-else:
-    # This is a new patch version.
+
+if len(all_tags) == 0:
+    # No tags besides the most recent one were found, so this is a new
+    # patch version. So "increase" the patch version:
```

Comment:
I honestly don't remember the exact logic here, so I can't comment on the correctness. Just make sure you run this on all series branches 3.7 and up and it behaves correctly. I know 3.8 and others aren't needed, but just for verification.